crates/solvers/Cargo.toml

```diff
@@ -24,6 +24,8 @@ hex = "0.4"
 hyper = "0.14"
 itertools = "0.11"
 num = "0.4"
+prometheus = { workspace = true }
+prometheus-metric-storage = { workspace = true }
```

Comment:
My plan didn't pan out entirely. 🙁 
Maybe we can reexport the types we need from those crates in the `observe` crate as well in another PR. Then `observe` should be the only crate that directly depends on them.